rust_src/src/hashtable.rs

```diff
@@ -0,0 +1,42 @@
+use lisp::{LispObject, ExternalPtr};
+use remacs_sys::{Lisp_Hash_Table, PseudovecType};
+use std::mem;
+
+pub type LispHashTableRef = ExternalPtr<Lisp_Hash_Table>;
+
+impl LispHashTableRef {
+    pub fn allocate() -> LispHashTableRef {
+        let vec_ptr =
+            allocate_pseudovector!(Lisp_Hash_Table, count, PseudovecType::PVEC_HASH_TABLE);
+        LispHashTableRef::new(unsafe { mem::transmute(vec_ptr) })
+    }
+
+    pub fn copy(&mut self, other: &LispHashTableRef) {
+        *self = *other
```

Comment:
Looking at this again, I don't believe this is what I want to be doing. I don't believe this is copying the structs into each other.